docs/RFCs/delivering-different-features.md

```diff
@@ -0,0 +1,78 @@
+# Building and delivering different features 
+
+## Context
+
+Our current build system is hard coded based on the profiling / telemetry requirements.
+With the growth of libdatadog, we need to adjust our build system.
+
+The current commands are :
+
+```
+./build_profiling-profiling.sh <package_directory>
+```
+
+OR
+
+```
+./build_telemetry-ffi.sh <package_directory>
+```
+
+Releases are then created through CI pipelines.
+
+The issue is that we have several features:
+- telemetry
+- exporting profiles
+- aggregating profiles
+- crash tracking
+- symbolization (ongoing).
+
+Not every profiler / tracer needs every feature, so we need it here way of Building and delivering exactly what we need.
+
+## Not a solution
+
+### Delivering different libraries
+
+By delivering features in several libraries, we expose ourselves to duplicated Rust runtimes (with possibly different versions and behaviours).
```

Comment:
While it's clear to me what you mean by "delivering different libraries" here, I'm not sure it'll be clear for all readers that you mean different libraries per-feature (telemetry, profiling, ...) or even finer-grained VS different libraries among other axis (e.g. for each language).